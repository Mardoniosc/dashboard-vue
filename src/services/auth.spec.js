import mockAxios from 'axios'
import AuthService from './auth'

jest.mock('axios')

describe('AuthService', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('Deve retornar um token quando usuário logar', async () => {
    const token = '654.654.654'
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } })
    })

    const response = await AuthService(mockAxios).login({ email: 'algo@live.com', password: '123321' })
    expect(response.data).toHaveProperty('token')
    expect(response).toMatchSnapshot()
  })

  it('Deve retornar um usuário quando o usuário se registrar', async () => {
    const user = {
      name: 'Joao',
      password: '147258',
      email: 'joao@gmail.com'
    }

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: user })
    })

    const response = await AuthService(mockAxios).register(user)
    expect(response.data).toHaveProperty('name')
    expect(response.data).toHaveProperty('password')
    expect(response.data).toHaveProperty('email')
    expect(response.data).toMatchSnapshot()
  })

  it('Deve retornar um erro de não encontrado', async () => {
    const errors = { status: 404, statusText: 'Not Found' }

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ request: errors })
    })

    const response = await AuthService(mockAxios).login({ email: 'algo@live.com', password: '123321' })
    expect(response.errors).toHaveProperty('status')
    expect(response.errors).toHaveProperty('statusText')
  })
})
