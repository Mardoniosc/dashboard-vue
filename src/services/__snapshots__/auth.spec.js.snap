// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthService Deve retornar um token quando usuário logar 1`] = `
Object {
  "data": Object {
    "token": "654.654.654",
  },
  "errors": null,
}
`;

exports[`AuthService Deve retornar um usuário quando o usuário se registrar 1`] = `
Object {
  "email": "joao@gmail.com",
  "name": "Joao",
  "password": "147258",
}
`;
